src/workspaces/SettingsModal/SettingsModal.tsx

```diff
@@ -0,0 +1,176 @@
+import { ButtonPrimary, Modal, SpinnerOverlay } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import React, { ReactNode, useEffect, useState } from 'react';
+import { Ajax } from 'src/libs/ajax';
+import { withErrorReporting } from 'src/libs/error';
+import Events, { extractWorkspaceDetails } from 'src/libs/events';
+import { isFeaturePreviewEnabled } from 'src/libs/feature-previews';
+import { GCP_BUCKET_LIFECYCLE_RULES } from 'src/libs/feature-previews-config';
+import { useCancellation } from 'src/libs/react-utils';
+import BucketLifecycleSettings from 'src/workspaces/SettingsModal/BucketLifecycleSettings';
+import {
+  BucketLifecycleSetting,
+  DeleteBucketLifecycleRule,
+  isBucketLifecycleSetting,
+  isDeleteBucketLifecycleRule,
+  modifyFirstBucketDeletionRule,
+  removeFirstBucketDeletionRule,
+  suggestedPrefixes,
+  WorkspaceSetting,
+} from 'src/workspaces/SettingsModal/utils';
+import { isOwner as isWorkspaceOwner, WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+interface SettingsModalProps {
+  workspace: Workspace;
+  onDismiss: () => void;
+}
+
+/**
+ * This modal is assumed to only be displayed for Google workspaces.
+ */
+const SettingsModal = (props: SettingsModalProps): ReactNode => {
+  const { namespace, name } = props.workspace.workspace;
+  const isOwner = isWorkspaceOwner(props.workspace.accessLevel);
+
+  const [lifecycleRulesEnabled, setLifecycleRulesEnabled] = useState(false);
+  const [prefixes, setPrefixes] = useState<string[]>([]);
+  const [lifecycleAge, setLifecycleAge] = useState<number | null>(null);
+
+  // Original settings from server, may contain multiple types
+  const [workspaceSettings, setWorkspaceSettings] = useState<WorkspaceSetting[] | undefined>(undefined);
+
+  const signal = useCancellation();
+
+  const getFirstBucketLifecycleSetting = (
+    settings: WorkspaceSetting[],
+    printMultipleWarning = false
+  ): BucketLifecycleSetting | undefined => {
+    const bucketLifecycleSettings: BucketLifecycleSetting[] = settings.filter((setting: WorkspaceSetting) =>
+      isBucketLifecycleSetting(setting)
+    ) as BucketLifecycleSetting[];
+    if (bucketLifecycleSettings.length > 0) {
+      if (printMultipleWarning && bucketLifecycleSettings.length > 1) {
+        // eslint-disable-next-line no-console
+        console.log('Multiple bucket lifecycle settings found, displaying only first');
```

Comment:
There should only be a single bucket lifecycle setting per workspace. I think this is still fine, but just calling out that this is an unexpected state for a workspace to be in.

Reply:
Since the format of the API is an array of settings, I figure that I do need to at least handle the unexpected case with a log message.